apps/emqx/test/emqx_broker_SUITE.erl

```diff
@@ -559,45 +559,70 @@ t_connected_client_count_transient_takeover(Config) when is_list(Config) ->
     %% we spawn several clients simultaneously to cause the race
     %% condition for the client id lock
     NumClients = 20,
+    ConnectSuccessCntr = counters:new(1, []),
+    ConnectFailCntr = counters:new(1, []),
+    ConnectFun =
+        fun() ->
+            try
+                process_flag(trap_exit, true),
+                {ok, ConnPid} =
+                    emqtt:start_link([
+                        {clean_start, true},
+                        {clientid, ClientID}
+                        | Config
+                    ]),
```

Comment:
Does it make sense to move it out of try expression? So that it's indeed only a connect call left inside.

Reply:
I think we want to catch if the `emqtt:start_link` fails as well and treat it as a connect failure but the `process_flag` call can be moved out from the try block.

Reply:
Moved the process_flag call up one level and force pushed